src/Transformers/DataTransformer.php

```diff
@@ -199,12 +199,16 @@ protected function resolvePropertyValue(
             return null;
         }
 
-        if (is_array($value) && ($trees->only instanceof AllTreeNode || $trees->only instanceof PartialTreeNode)) {
-            $value = Arr::only($value, $trees->only->getFields());
-        }
+        if (is_array($value)) {
```

Comment:
Could you rewrite these like they were, not nested if's?

+ Is it possible to add a test which checks this? I'm certain I or someone else will change this logic in the future and I don't want it to break again.